src/WidgetMessaging.js

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2017 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import dis from './dispatcher';
+
+let listenerCount = 0;
+let messageEndpoints = [];
+
+/**
+ * Handle widget postMessage events
+ * @param  {Event} event Event to handle
+ * @return {undefined}
+ */
+function onMessage(event) {
```

Comment:
This should probably always return something back to the widget, like the scalar API. 

Ie:
```
{ success: true, .. other params .. }
// other params would be the original request
```

```
{ success: false, error: "Unknown action", .. other params .. }
```

Reply:
Thanks for pointing this out @turt2live - I've added responses (and comments documenting the API) to bring it in line with the scalarMessaging API (slightly different structure to the one that you describe above).